lib/api/core/models/security/role.js

```diff
@@ -38,207 +35,147 @@ const
-      promises.push(executeClosure(this, this[_kuzzle], path, actionRights, request));
-    } else {
-      return Bluebird.reject(errorsManager.getError(
-        'invalid_rights_given',
-        this._id, path.join('/'),
-        actionRights));
+    else {
+      return false;
     }
 
-    promises.push(checkRestrictions(request, restrictedTo));
+    if (typeof actionRights !== 'boolean' || !actionRights) {
+      return false;
+    }
 
-    return Bluebird.all(promises)
-      .then(results => results.every(r => r));
+    return checkRestrictions(request, restrictedTo);
   }
 
   /**
-   * @returns {Promise.<boolean>}
+   * @returns {Promise}
    */
-  validateDefinition () {
-    if (typeof this.controllers !== 'object' || this.controllers === null) {
-      return Bluebird.reject(errorsManager.getError('controllers_definition_not_an_object'));
+  async validateDefinition () {
+    if (!_.isPlainObject(this.controllers)) {
+      errorsManager.throw('controllers_definition_not_an_object');
     }
+
     if (Object.keys(this.controllers).length === 0) {
-      return Bluebird.reject(errorsManager.getError('empty_controllers_definition'));
+      errorsManager.throw('empty_controllers_definition');
     }
 
-    return new Bluebird((resolve, reject) => {
-      const promises = [];
-
-      const result = Object.keys(this.controllers).every(controllerKey => {
-        const controllerRights = this.controllers[controllerKey];
+    Object.entries(this.controllers).forEach(this.validateControllerRights);
+  }
 
-        if (typeof controllerRights !== 'object' || controllerRights === null) {
-          reject(errorsManager.getError(
-            'controller_definition_not_an_object',
-            controllerKey));
-          return false;
-        }
-        if (Object.keys(controllerRights).length === 0) {
-          reject(errorsManager.getError(
-            'empty_controller_definition',
-            controllerKey));
-          return false;
-        }
-        if (controllerRights.actions === undefined) {
-          reject(errorsManager.getError(
-            'actions_attribute_missing_in_controller_definition',
-            controllerKey));
-          return false;
-        }
-        if (typeof controllerRights.actions !== 'object' || controllerRights.actions === null) {
-          reject(errorsManager.getError(
-            'actions_attribute_not_an_object_in_controller_definition',
-            controllerKey));
-          return false;
-        }
-        if (Object.keys(controllerRights.actions).length === 0) {
-          reject(errorsManager.getError(
-            'actions_attribute_empty_in_controller_definition',
-            controllerKey));
-          return false;
-        }
+  /**
+   * Verifies that a controller rights definition is correct
+   *
+   * @param  {Array.<string, Object>}
+   * @throws If the controller definition is invalid
+   */
+  validateControllerRights ([name, controller]) {
```

Comment:
Open discussion/Debate: 
I'm not comfortable with destructuring array in method arguments. Maybe you can destructur the array before passing it instead ? 
```
Object.entries(this.controllers).forEach(...this.validateControllerRights);
```

Reply:
Sure, why not. I have no strong feelings about this.